libraries/kiwi/kiwi-triplestore/src/main/java/org/apache/marmotta/kiwi/sail/KiWiValueFactory.java

```diff
@@ -565,6 +565,8 @@ public Statement createStatement(Resource subject, URI predicate, Value object,
 
             KiWiTriple result = new KiWiTriple(ksubject,kpredicate,kobject,kcontext);
 
+            boolean needsDBLookup = false;
+
             synchronized (registry) {
```

Comment:
Rather than synchronized we could also use a Lock in the registry. This would avoid having to use a boolean flag to indicate if a DB lookup is needed (we could release the lock before reading from the DB).